### ISSUE TITLE
Whats the difference from react-storybook?

### ISSUE THREAD

**Issue by gajus** [react-cosmos/react-cosmos#233]
https://github.com/storybooks/react-storybook has existed for a while now.

I haven't used either beyond a sandbox experiment.

What is the difference between the two?

A comparison table would be useful.

**Comment by skidding** [comment]
Not that it means much, but Cosmos started 2y before Storybook. A comparison table would be nice, and we should probably throw others in the mix as well, like [Carte Blanche](https://github.com/carteb/carte-blanche) or [react-styleguidist](https://github.com/styleguidist/react-styleguidist).

I mildly touch this subject in [an article I wrote last month](https://medium.com/@skidding/fighting-for-component-independence-2a762ee53272). You mind find it interesting.

> [...]  I’m a fan of Storybook’s friendly “stories” and Nik Graf’s idea to generate random props, yet I find React Cosmos unique in its quest to surface the relation between raw input data and rich component output. [...]

In short, these are the things I've not seen in any other tool:
- The "fixture editor" embedded in the UI, which allows you to quickly fiddle with input data and see state changes, making the experience interactive
- The support for mocking component state (not just props), more recently Redux state as well

Hope this helps.

**Comment by gajus** [comment]
> Not that it means much, but Cosmos started 2y before Storybook.

I see. My impression was that react-cosmos is a new project. I saw it first few months ago on HN.

> A comparison table would be nice, and we should probably throw others in the mix as well, like Carte Blanche or react-styleguidist.

Good idea. You can use https://npmcompare.com/compare/connect,express to start.

> Hope this helps.

Thats very useful. Thank you

**Comment by skidding** [comment]
npmcompare is interesting, but it doesn't work so well with Cosmos and tools alike since they are composed of many npm packages without a designated "main" one. It might be the CLI, or the UI module, and at least in Cosmos' case the package names changed a few times along the years.

I'm currently working on other PRs ([e.g.](https://github.com/react-cosmos/react-cosmos/issues/232)), so I'll leave this open for now for anyone else who's asking the same question or wants to help document this better.

**Comment by skidding** [comment]
Added a [README section](https://github.com/react-cosmos/react-cosmos/tree/1bd42d2104b8e29241d0d8a5334b80e36f169682#why-cosmos-and-not-x) on this.
